src/modules/App.tsx

```diff
@@ -1,117 +1,93 @@
-import * as React from 'react';
-import Button from '../components/Button';
-import Input from '../components/Input';
-import {
-  logIn,
-  resetForm,
-  saveTextInput,
-} from '../redux/actions';
-import '../styles/app.scss';
-import { setPassword } from "../password";
-import { connect } from 'react-redux'
-import { StoreState } from '../redux/reducers';
+import React from "react";
+import { connect } from "react-redux";
 
-interface AppProps {
-  history?: { action: string }
-  text: string;
-  logIn: typeof logIn;
-  resetForm: typeof resetForm;
-  saveTextInput: typeof saveTextInput,
-}
+import { Button, Input } from "components";
+// you already imported this in index.tsx
+// import "../styles/app.scss";
+import { logIn, resetForm, saveTextInput } from "redux/actions";
+// this isn't a great name. it implies an action is being taken whereas this is just a constant
+import { setPassword } from "../password";
+import { AppProps, AppState, StoreState } from "./types";
 
-class App extends React.Component<AppProps> {
+// create interfaces for your state
+class App extends React.Component<AppProps, AppState> {
   state = {
-    isErrorMsgShown: false,
-    password: "",
-  }
+    // avoid using shorthand (msg). be as explicit as possible. I changed this to a string so that you can more easily display a variety of error messages
+    errorMessage: "",
+  };
 
-  onButtonClick = (e: React.ChangeEvent<HTMLFormElement>) => {
-    const { resetForm } = this.props;
-    if (e.target.type === "submit") {
-      this.onSubmit(e);
-    } else {
-      this.setState({ isErrorMsgShown: false, password: "" })
-      resetForm()
-    };
-  }
+  // you should always strive to have a separation of concerns. a function should do one thing.
+  reset = () => {
```

Comment:
so you generally do the es6 destructuring thing for your props when you have two or more?